### ISSUE TITLE
Newest release breaks links for sites in subfolder

### ISSUE THREAD

**Issue by MattyBalaam** [jonom/silverstripe-betternavigator#25]
In 2.1 if I had a site in e.g. locahost/foo, if I click on any of the links this is run on the subfolder. However on 3.0 it removes this, so instead of e.g. localhost/foo/dev/build it gets localhost/dev/build.

**Comment by MattyBalaam** [comment]
On the page the <base> tag is 

<base href="http://localhost/foo/"

**Comment by MattyBalaam** [comment]
```
$nav['CMSLink']['Link']
```
result: **admin/pages/edit/show/1**
````
Controller::join_links('/', $nav['CMSLink']['Link'])
````
result: **/admin/pages/edit/show/1**
```
Controller::join_links($nav['CMSLink']['Link']) 
```
result: **admin/pages/edit/show/1**

**Comment by jonom** [comment]
Okay, I apparently misunderstood how the base tag works. That change will need to be reverted or a patch developed.

**Comment by MattyBalaam** [comment]
Sorry, I have investigated a little more, and it may be the non-standard .htaccess file used on this particular site. Reverting to the default has solved the issue.

**Comment by jonom** [comment]
Yeah I tried the config setting I suggested for you and didn't get the result I expected at all. I might try reverting the commit above and converting all the links to be absolute, hopefully that will work out of the box for your scenario as well as sites without a base tag.

**Comment by quamsta** [comment]
Any word on a fix for this? Very irritating in our dev environments

**Comment by MattyBalaam** [comment]
How about using the older version until this is fixed?

**Comment by jonom** [comment]
Fixed by #28 gracias @3Dgoo

**Comment by quamsta** [comment]
Thank you! <3
